### PR TITLE
Feature/version validation

### PR DESCRIPTION
**Pull request by kfwilson**
on first run, saves environment to requirements.txt. on subsequent runs, compares current environment to original. If mismatches, you'll see an error like the following:

image

### CODE REVIEW

File: src/vivarium_cluster_tools/utilities.py [hunk 1]
```diff
@@ -7,6 +7,11 @@
 import sys
 
 from loguru import logger
+from typing import Dict, List, Tuple
+try:
+    from pip._internal.operations import freeze
```
Comment (collijk) [inline src/vivarium_cluster_tools/utilities.py]: Document why you need this try/except

File: src/vivarium_cluster_tools/utilities.py [hunk 2]
```diff
@@ -157,3 +162,72 @@ def wrapped(*args, **kwargs):
...
+                         f'In order to successfully run, you should make a new environment using the requirements.txt '
+                         f'file found in the output directory. Differences found as follows: {differences}.')
+
+
+def validate_environment(output_dir: Path):
+    original_environment_file = output_dir / 'requirements.txt'
+
+    current_environment_list = [p for p in freeze.freeze()]
+    if not original_environment_file.exists():  # original run
+        with open(original_environment_file, 'w') as f:
```
Comment (collijk) [inline src/vivarium_cluster_tools/utilities.py]: With path.open('w') as f

**Comment by collijk** [comment]
Looks great!  Maybe add some logging?

File: src/vivarium_cluster_tools/utilities.py (hunk 2 above)
Comment (kfwilson) [reply src/vivarium_cluster_tools/utilities.py]: done

File: src/vivarium_cluster_tools/utilities.py (hunk 1 above)
Comment (kfwilson) [reply src/vivarium_cluster_tools/utilities.py]: added comment explaining where freeze is depends on version of pip

**Pull request merged**